webviz_subsurface/_abbreviations/reservoir_simulation.py

```diff
@@ -30,47 +30,62 @@ def simulation_vector_base(vector: str) -> str:
+    return vector.split(":", 1)[0].split("_")[0]
 
 
 def simulation_vector_description(vector: str) -> str:
     """Returns a more human friendly description of the simulation vector if possible,
      otherwise returns the input as is.
     """
     [vector_name, node] = vector.split(":", 1) if ":" in vector else [vector, None]
+    fip = None
+    comp = None
     if len(vector_name) == 8:
-        # Region vectors for other FIP regions than FIPNUM are written on a special form:
-        # 8 signs, with the last 3 defining the region.
-        # E.g.: For an array "FIPREG": ROIP is ROIP_REG, RPR is RPR__REG and ROIPL is ROIPLREG
-        # Underscores _ are always used to fill
-        [vector_base_name, fip] = [vector_name[0:5].rstrip("_"), vector_name[5:]]
-        try:
-            if SIMULATION_VECTOR_TERMINOLOGY[vector_base_name]["type"] == "region":
-                vector_name = vector_base_name
-            else:
-                fip = None
-        except KeyError:
-            fip = None
-    else:
-        fip = None
+        if vector_name[0] == "R":
+            # Region vectors for other FIP regions than FIPNUM are written on a special form:
+            # 8 signs, with the last 3 defining the region.
+            # E.g.: For an array "FIPREG": ROIP is ROIP_REG, RPR is RPR__REG and ROIPL is ROIPLREG
+            # Underscores _ are always used to fill
+            [vector_base_name, fip] = [vector_name[0:5].rstrip("_"), vector_name[5:]]
+            try:
+                if SIMULATION_VECTOR_TERMINOLOGY[vector_base_name]["type"] == "region":
+                    vector_name = vector_base_name
+            except KeyError:
+                pass
+        elif vector_name.startswith(("W")) and vector_name[4] == "L":
+            # These are completion vectors, e.g. WWCTL:__1:OP_1 and WOPRL_10:OP_1 for
+            # water-cut in OP_1 completion 1 and oil production rate in OP_1 completion 10
+            [vector_base_name, comp] = [vector_name[0:5], vector_name[5:].lstrip("_")]
+            try:
+                if (
+                    SIMULATION_VECTOR_TERMINOLOGY[vector_base_name]["type"]
+                    == "completion"
+                ):
+                    vector_name = vector_base_name
+            except KeyError:
+                pass
+
     if vector_name in SIMULATION_VECTOR_TERMINOLOGY:
         metadata = SIMULATION_VECTOR_TERMINOLOGY[vector_name]
         description = metadata["description"]
         if node is not None:
             description += (
-                f", {metadata['type'].replace('_', ' ')} {fip} {node}"
-                if fip is not None
-                else f", {metadata['type'].replace('_', ' ')} {node}"
+                f", {metadata['type'].replace('_', ' ')} {node}"
+                if fip is None and comp is None
+                else f", {metadata['type'].replace('_', ' ')} {fip} {node}"
+                if comp is None
+                else f", well {node} completion {comp}"
```

Comment:
This starts to become complex to read. Maybe we could utilize that in earlier `if/else` branches of this function, we know when we have `fip` or `comp` defined, and instead do early returns?

Reply:
We could. Didn't shuffle more code than I had too for this PR, but I agree that it is getting a bit messy.